bench/syclblas/syclblas_benchmark_level2.cpp

```diff
@@ -23,32 +23,33 @@
  *
  **************************************************************************/
 
-#include "../blas_benchmark2.hpp"
+#include "../common/blas_benchmark.hpp"
 
 #include <interface/blas2_interface.hpp>
 
 using namespace blas;
 
-BENCHMARK_NAME_FORMAT(blas_level_2) {
+BENCHMARK_NAME_FORMAT(syclblas_level_2) {
   std::ostringstream fname;
-  fname << name() << "_" << std::get<0>(params) << "_" << std::get<1>(params)
-        << "_" << std::get<2>(params);
+  fname << typeid(ElemT).name() << "_" << name() << "_" << std::get<0>(params)
```

Comment:
This introduces a dependency on RTTI for the benchmark, can we avoid it?

Reply:
I *think* we can. I've thrown together a solution that uses function template specialisation instead, as follows: 

```
// blas_benchmark.hpp
template <typename T>
const inline std::string type_string() {
  return "unknown";
}

template <>
const inline std::string type_string<float>() {
  return "float";
}


template <>
const inline std::string type_string<double>() {
  return "double";
}

// ...
// Inside the benchmark struct
template<typename T>
static std::string typestr() { 
  return type_string<T>();
}
```

Which we use as follows in (say) `syclblas_benchmark_level2.cpp`:

```
BENCHMARK_NAME_FORMAT(syclblas_level_2) {
  std::ostringstream fname;
  fname << benchmark<>::typestr<ElemT>() << "_" << name() << "_" << std::get<0>(params)
        << "_" << std::get<1>(params) << "_" << std::get<2>(params);
  return fname.str();
}
```
It gives us the functionality we want (being able to distinguish between benchmarks based on the scalar type), and *should* decide on the type string statically.